### PR TITLE
Add NCCL workload for A3 mega and update README guides.

### PR DESCRIPTION
**Pull request by samcmho**

### CODE REVIEW

File: sample_workloads/nccltest/a3-mega/README.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+    16777216       4194304     float     sum      -1   1932.8    8.68   16.28    N/A   2014.4    8.33   15.62    N/A
+    33554432       8388608     float     sum      -1   2190.8   15.32   28.72    N/A   2395.8   14.01   26.26    N/A
+    67108864      16777216     float     sum      -1   2367.6   28.34   53.15    N/A   2389.8   28.08   52.65    N/A
+   134217728      33554432     float     sum      -1   3539.5   37.92   71.10    N/A   3266.7   41.09   77.04    N/A
+   268435456      67108864     float     sum      -1   5969.3   44.97   84.32    N/A   5850.8   45.88   86.03    N/A
+   536870912     134217728     float     sum      -1    11625   46.18   86.59    N/A    11737   45.74   85.77    N/A
+  1073741824     268435456     float     sum      -1    23144   46.39   86.99    N/A    38777   27.69   51.92    N/A
+  2147483648     536870912     float     sum      -1    45662   47.03   88.18    N/A    45522   47.17   88.45    N/A
+  4294967296    1073741824     float     sum      -1    90227   47.60   89.25    N/A    90354   47.53   89.13    N/A
+  8589934592    2147483648     float     sum      -1   179880   47.75   89.54    N/A   178867   48.02   90.05    N/A
```
Comment (Chris113113) [inline sample_workloads/nccltest/a3-mega/README.md:70]: nit: These are a3 logs.

**Review by Chris113113 — APPROVED** [review]

File: sample_workloads/nccltest/a3-mega/README.md [hunk 2]
```diff
@@ -0,0 +1,148 @@
+# TCPX NCCL Level Benchmarks
+
+This document will walk you through how to build, push, and deploy the nccl-benchmark image for use on a GKE cluster configured with an A3 nodepool.
```
Comment (anthonyhan2) [inline sample_workloads/nccltest/a3-mega/README.md:3]: A3+?

File: sample_workloads/nccltest/a3-mega/README.md [hunk 3]
```diff
@@ -0,0 +1,148 @@
+# TCPX NCCL Level Benchmarks
```
Comment (anthonyhan2) [inline sample_workloads/nccltest/a3-mega/README.md:1]: Maybe use the external facing name for the A3+ transport?

File: sample_workloads/nccltest/a3-mega/docker/scripts/mpi_entry.sh
Comment (anthonyhan2) [inline sample_workloads/nccltest/a3-mega/docker/scripts/mpi_entry.sh:1]: Rename anything that says TCPX?

File: sample_workloads/nccltest/a3-mega/README.md (hunk 3 above)
Comment (samskillman) [reply sample_workloads/nccltest/a3-mega/README.md:1]: +1, "GPUDirect-TCPXO"

File: sample_workloads/nccltest/a3-mega/README.md (hunk 2 above)
Comment (samskillman) [reply sample_workloads/nccltest/a3-mega/README.md:3]: Yeah, "A3 Mega"

**Review by samskillman — APPROVED** [review]
@samcmho we should change all the "fastrak" to "tcpxo", but would appreciate input from @anthonyhan2 on whether the `pkill` command needs to still target the "fastrak" process.  After addressing those, LGTM.

**Comment by anthonyhan2** [comment]
I can't guarantee things won't break if we just change all fastrak to tcpxo, so I would highly recommend @samcmho to test it out first. The pkill command probably shouldn't change anytime soon.

**Review by anthonyhan2 — APPROVED** [review]

**Pull request merged**